copy/copy.go

```diff
@@ -386,6 +393,10 @@ func (c *copier) copy(ctx context.Context, src, srcComponents, target string, ov
 		if err := copyXAttrs(target, src, c.xattrErrorHandler); err != nil {
 			return errors.Wrap(err, "failed to copy xattrs")
 		}
+	} else if restoreFileInfo && targetFi != nil {
+		if err := c.copyFileInfo(targetFi, target); err != nil {
```

Comment:
Could we separate out the timestamp update and only call that. Performance isn't the main issue, but I think it makes the code easier to follow if it is clear in here that we call this just to restore the timestamp.